### DISCUSSION TITLE
MQTT Connection closed, is this a bug?

### DISCUSSION THREAD

**Discussion by perezd** [aqualinkd/AqualinkD#200] [General]
I've got a Pi Zero + the recommend USB cord from the wiki installed and working (website comes up as expected!). I've noticed in the logs this pattern:

```
Notice:  AqualinkD: Listening to Aqualink RS8 on serial port: /dev/ttyUSB0
Notice:  AqualinkD: Waiting for Control Panel probe
Notice:  NetService:Starting web server on port 80
Notice:  NetService:Starting MQTT client to homebridge.local:1883
Warning: NetService:MQTT Connection closed
```

I have an MQTT broker running on homebridge.local @ that port, I've check via MQTTExplorer AND mosquitto_sub from the RPI0 directly that it can communicate and hear updates to the `aqualinkd` topic.

What I'm wondering is, because this is not fully wired into a panel yet, is that why it's disconnecting MQTT? I'm trying to get everything ready before I install the new panel.

Thoughts?

**Comment by sfeakes** [comment]
It's probably because homebridge.local isn't resolving on your network.   .local is usually used for MDNS, and by default a Pi doesn't use MDNS, it uses DNS.  you can check this by telnet/ssh to the pi running AqualinkD and using the command `ping homebridge.local` see if that resolves and you get an a reply or error.
Without getting into DNS / MDNS on your network that will be dependent on many things, simply use the IP address rather than homebridge.local in the config.  If that works, then you can track down the other DNS issues at a later date.

EDIT.  Is it connecting and working then sometimes disconnect / reconnect, or you get constant connect / disconnect?   My reply was based on constant disconnect and aqualinkd never actually sending messages.   But either way, using the actual IP over a machine name would be the first thing to test.

**Reply by perezd** [reply]
Not only can I ping it, I can connect and listen to homebridge.local via mosquitto_sub. This is working as expected.

**Reply by sfeakes** [reply]
Does AqualinkD connect and send messages then disconnect, or is it not sending anything at all? (you’d need AqualinkD to also connect to the control panel for it to send any MQTT messages).   The disconnect are not normal.   But it could be for a number of reasons, and I’m trying to see if it is connecting at all, or in a constant state of trying to connect.  Using IP would be first thing to test though, as resolving can fail for a number of reasons and at different layers of the OS.

**Reply by perezd** [reply]
So it's not connected yet (pre-setting up hardware, pending controller install), so that's what I was wondering, is it disconnecting because it can't talk to the panel yet?

**Reply by sfeakes** [reply]
I just looked over the code. I could see where if AqualinkD does not get an initial probe from the panel, then it wouldn’t get to part of the code that does an MQTT ping, so depending on MQTT server config, that could close the connection thinking it’s dead.   So yes I could see this happening, but hard to say if that’s your case or not without more of the logs.  Do you know how long it is between connect and disconnect? if it’s instant This probably isn’t it.   You could also turn on debug logging and that might tell you more, if you want to play before connecting it to the control panel.

**Reply by perezd** [reply]
it doesn't seem instant, it connects and then disconnects roughly a second or two later. I suspect this is a lack of panel connectivity, everything else appears to be working fine. I'll report back once it's installed!

**Reply by perezd** [reply]
OK Panel installed and I can communicate with the panel via the aqualinkd web page so it's working as expected there! Now, I am still seeing this in the logs, no other errors:

```
Mar-20-23 13:14:48 PM Notice:  NetService:Starting MQTT client to homebridge.local:1883
Mar-20-23 13:14:48 PM Warning: NetService:MQTT Connection closed
```

Gonna try IP address here and see if DNS resolving isn't working as expected.

**Reply by perezd** [reply]
Confirmed, using IP works as expected. Nice work with this!!
